metrics/scaling/k8s_scale_net.sh

```diff
@@ -0,0 +1,288 @@
+	local n_pods=$2
+
+	info "And grab some stats"
+
+	local date_json="$(cat << EOF
+			"date": {
+				"ns": $(date +%s%N),
+                                "Date": "$(date -u +"%Y-%m-%dT%T.%3N")"
+                        }
+EOF
+        )"
+	metrics_json_add_array_fragment "$date_json"
+
+        local pods_json="$(cat << EOF
+	                "n_pods": {
+				"Result": ${n_pods},
+                                "Units" : "int"
+                        }
+EOF
+        )"
+        metrics_json_add_array_fragment "$pods_json"
+
+	local launch_json="$(cat << EOF
+                        "launch_time": {
+                                "Result": $launch_time_ms,
+                                "Units" : "ms"
+                        }
+EOF
+        )"
+        metrics_json_add_array_fragment "$launch_json"
+
+	info "launch [$launch_time_ms]"
+
+	metrics_json_close_array_element
+}
+
+init() {
+        info "Initialising"
+
+	local cmds=("bc" "jq")
+	check_cmds "${cmds[@]}"
+
+	info "Checking Kubernetes accessible"
+	local worked=$( kubectl get nodes > /dev/null 2>&1 && echo $? || echo $? )
+	if [ "$worked" != 0 ]; then
+		die "kubectl failed to get nodes"
+	fi
+
+	info $(get_num_nodes) "Kubernetes nodes in 'Ready' state found"
+
+	k8s_api_init
+
+	 # And now we can set up our results storage then...
+	 metrics_json_init "k8s"
+	 save_config
+}
+
+save_config() {
+	metrics_json_start_array
+
+	local json="$(cat << EOF
+	{
+		"testname": "${TEST_NAME}",
+                "NUM_PODS": ${NUM_PODS},
+                "STEP": ${STEP},
+                "wait_time": ${wait_time},
+                "delete_wait_time": ${delete_wait_time},
+                "settle_time": ${settle_time}
+        }
+EOF
+)"
+        metrics_json_add_array_element "$json"
+	metrics_json_end_array "Config"
+}
+
+run() {
+	info "Running test"
+	local header_post="Content-Type: application/json"
+	local header_patch="Content-Type: application/strategic-merge-patch+json"
+	local base_curl=${API_ADDRESS}:${API_PORT}/apis/apps/v1/namespaces/default/deployments
+
+	trap cleanup EXIT QUIT KILL
+
+	metrics_json_start_array
+
+	for reqs in $(seq ${STEP} ${STEP} ${NUM_PODS}); do
+		info "Testing replicas ${reqs} of ${NUM_PODS}"
+		# Generate the next yaml file
+
+		local runtime_command
+		if [ -n "$RUNTIME" ]; then
+			runtime_command="s|@RUNTIMECLASS@|${RUNTIME}|g"
+		else
+			runtime_command="/@RUNTIMECLASS@/d"
+		fi
+
+		local input_template
+		local generated_file
+		if [ "$use_api" != "no" ]; then
+			input_template=$input_json
+			generated_file=$generated_json
+		else
+			input_template=$input_yaml
+			generated_file=$generated_yaml
+		fi
+
+		sed -e "s|@REPLICAS@|${reqs}|g" \
+			-e $runtime_command \
+			-e "s|@DEPLOYMENT@|${deployment}|g" \
+			-e "s|@LABEL@|${LABEL}|g" \
+			-e "s|@LABELVALUE@|${LABELVALUE}|g" \
+			-e "s|@GRACE@|${grace}|g" \
+			< ${input_template} > ${generated_file}
+
+		info "Applying changes"
+		local start_time=$(date +%s%N | cut -b1-13)
```

Comment:
nit: I'm not a fan of this `cut`ing of the ns timestamp - it looks like it hard wires the digits, and that might not scale over time.... there is already a ms method over in the json script to grab the ms timestamp if you need it - see https://github.com/clearlinux/cloud-native-setup/blob/master/metrics/lib/json.bash#L20

Reply:
New commit does not `cut` the ns timestamp.